index.js

```diff
@@ -47,10 +42,85 @@ module.exports = {
   },
 
   forEach: function(path, cb, thisArg) {
-    forEach(path.match(SPLIT_REGEX), cb, thisArg)
+    forEach(split(path), cb, thisArg)
+  }
+}
+
+function setterEval(path) {
+  return (
+    setCache[path] ||
+    (setCache[path] = new Function(
+      'data, value',
+      expr(path, 'data') + ' = value'
+    ))
+  )
+}
+
+function setterFallbackMemoized(path) {
+  var parts = normalizePath(path)
+  return function(data, value) {
+    return setterFallback(parts, data, value)
```

Comment:
lets define this inline and save the additional function declaration (same with getter)